drivers/sensor/ti/ina23x/ina230.c

```diff
@@ -185,13 +193,44 @@ static int ina230_calibrate(const struct device *dev)
 static int ina230_init(const struct device *dev)
 {
 	const struct ina230_config *const config = dev->config;
+	struct ina230_data *data = dev->data;
+	uint16_t word = 0;
 	int ret;
 
 	if (!device_is_ready(config->bus.bus)) {
 		LOG_ERR("I2C bus %s is not ready", config->bus.bus->name);
 		return -ENODEV;
 	}
 
+	data->bus_voltage_scaling = INA236_BUS_VOLTAGE_UV_LSB;
+	data->power_scaling = INA236_POWER_SCALING;
+
+	ret = ina23x_reg_read_16(&config->bus, INA236_REG_DEVICE_ID, &word);
+	if (ret < 0 || word != INA236_DEVICE_ID) {
+		LOG_INF("Failed to read device ID. Assuming INA230");
+		goto setup_dev_ina230;
+	}
+
+	ret = ina23x_reg_read_16(&config->bus, INA236_REG_MANUFACTURER_ID, &word);
+	if (ret < 0 || word != INA236_MANUFACTURER_ID) {
+		LOG_INF("Failed to read manufacturer ID. Assuming INA230");
+		goto setup_dev_ina230;
+	}
+
+	goto setup_dev;
+
+setup_dev_ina230:
```

Comment:
Prepare separate init functions based on the variant. You can follow the i.e https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/led/lp50xx.c approach.

Reply:
I was looking for this before I decided to go the "runtime" way. I hope I used the macro correctly in the updated version.